File: Java/1518_numWaterBottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int drankTotal = numBottles;
        int numEmpty = numBottles;
        while(numEmpty >= numExchange){
            drankTotal += numEmpty / numExchange;
            numEmpty = (numEmpty % numExchange) + (numEmpty / numExchange);
        }
            
        return drankTotal;
    }
}

File: Python/1518_numWaterBottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        drankTotal = numBottles
        numEmpty = numBottles
        while numEmpty >= numExchange:
            drankTotal += numEmpty // numExchange
            numEmpty = (numEmpty % numExchange) + (numEmpty // numExchange)
            
        return drankTotal
